src/rotary-dialer-element.ts

```diff
@@ -0,0 +1,297 @@
+    slots: SVGPathElement,
+    digit: Digit,
+    direction: Direction = Direction.normal,
+    duration = '1000ms',
+    timingFn = 'ease-out'
+  ) {
+    switch (digit) {
+      case 0:
+        slots?.setAttribute('style', `animation: dialZero ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 1:
+        slots?.setAttribute('style', `animation: dialOne ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 2:
+        slots?.setAttribute('style', `animation: dialTwo ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 3:
+        slots?.setAttribute(
+          'style',
+          `animation: dialThree ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case 4:
+        slots?.setAttribute('style', `animation: dialFour ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 5:
+        slots?.setAttribute('style', `animation: dialFive ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 6:
+        slots?.setAttribute('style', `animation: dialSix ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case 7:
+        slots?.setAttribute(
+          'style',
+          `animation: dialSeven ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case 8:
+        slots?.setAttribute(
+          'style',
+          `animation: dialEight ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case 9:
+        slots?.setAttribute('style', `animation: dialNine ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      default:
+        slots?.removeAttribute('style');
+    }
+  }
+
+  private addDialerAnim(digit: Digit) {
+    this.handleRotation(digit);
+  }
+
+  private removeDialerAnim() {
+    const slots = this.shadowRoot?.querySelector('#slots');
+    slots?.classList.remove('dialer-anim');
+    slots?.removeAttribute('style');
+  }
+
+  private up(digit: Digit) {
+    this.dial(digit);
+  }
+
+  /**
+   * Exposed lazy dial by applying dial method with a given argument of number from 0-9
+   * @param digit
+   */
+  public dial(digit: Digit) {
+    this.removeDialerAnim();
+    this.addDialerAnim(digit);
+  }
+
+  private onValueChange(event: KeyboardEvent) {
+    const target = event.target as HTMLInputElement;
+    const digit = parseInt(target.value) as Digit;
+    this.dial(digit);
+    target.value = '';
+  }
+
+  private handleRotation(digit: Digit) {
+    const slots = this.shadowRoot?.querySelector('#slots') as SVGPathElement;
+    slots.onanimationend = () => {
+      this.dispatchEvent(
+        new CustomEvent('dial', {
+          detail: { digit },
+        })
+      );
+    };
+    const rAF = requestAnimationFrame(() => {
+      slots?.classList.add('dialer-anim');
+      // When you click on a digit, the circle-hole of that digit
+      // should go all the way until the finger stop.
+      RotaryDialerElement.matchDigit(slots, digit);
```

Comment:
Why make `matchDigit()` static? It requires repeating the class name whenever you use it...

Reply:
My code inspector (webstorm) always suggesting a potential pure util function to be static, I actually like it, we don't have any reference or use to the instance inside i, we're not gonna need it on the prototype as well :)